pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,263 @@
+		} else {
+			w.WriteHeader(http.StatusNoContent)
+		}
+	default:
+		w.Header().Set("Allow", "GET, POST, DELETE")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Unsupported method, supported methods are GET POST DELETE"})
+	}
+}
+
+// Fetch the current user with auth.User token and returning the user-setting ConfigMap.
+func (h *UserSettingsHandler) getUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := h.createServiceAccountClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return client.CoreV1().ConfigMaps(USER_SETTINGS_NAMESPACE).Get(context, resourceData.getConfigMapName(), meta.GetOptions{})
+}
+
+// Fetch the current user with auth.User token and create a new user-setting ConfigMap.
+// Returns the existing ConfigMap if it is already exist.
+func (h *UserSettingsHandler) createUserSettings(user *auth.User) (*core.ConfigMap, error) {
+	context := context.TODO()
+
+	resourceData, err := h.getResourceData(context, user)
+	if err != nil {
+		return nil, err
+	}
+
+	role := &rbac.Role{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "Role",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getRoleName(),
+		},
+		Rules: []rbac.PolicyRule{
+			rbac.PolicyRule{
+				APIGroups: []string{
+					"", // Core group, not "v1"
+				},
+				Resources: []string{
+					"configmaps", // Not "ConfigMap"
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"patch",
+					"update",
+					"watch",
+				},
+				ResourceNames: []string{
+					resourceData.getConfigMapName(),
+				},
+			},
+		},
+	}
+
+	roleBinding := &rbac.RoleBinding{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "RoleBinding",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getRoleBindingName(),
+		},
+		Subjects: []rbac.Subject{
+			rbac.Subject{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "User",
+				Name:     user.Username,
+			},
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     resourceData.getRoleName(),
+		},
+	}
+
+	configMap := &core.ConfigMap{
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: resourceData.getConfigMapName(),
+		},
+	}
+
+	client, err := h.createServiceAccountClient()
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = client.RbacV1().Roles(USER_SETTINGS_NAMESPACE).Create(context, role, meta.CreateOptions{})
+	if err != nil {
+		klog.Warningf("Could not create user-settings Role \"%s\" for user \"%s\" (UID: %s)", resourceData.getRoleName(), resourceData.Username, resourceData.UID)
+	}
+
+	_, err = client.RbacV1().RoleBindings(USER_SETTINGS_NAMESPACE).Create(context, roleBinding, meta.CreateOptions{})
```

Comment:
What happens when one of these resources(Role/Rolebinding/SA) fails to get created? Don't think we should proceed with creating the rest of them and leave it in an inconsistent state.

Reply:
Yep, was ignored before. I switched to this code to ignore just an `AlreadyExists` error. Everything else let the call fail and was logged then in `HandleUserSettings`.

```go
	if err != nil && !apierrors.IsAlreadyExists(err) {
		return nil, err
	}
```